lib/Braintree/OAuthGateway.php

```diff
@@ -61,7 +61,8 @@ private function _verifyGatewayResponse($response)
 
     public function _mapError($result)
     {
-        $error = $result->errors->deepAll()[0];
+        $errors = $result->errors->deepAll();
+        $error = $errors[0];
```

Comment:
The original code from Braintree does not make this check so I'm inclined to leave it out.